src/azure-cli/azure/cli/command_modules/storage/__init__.py

```diff
@@ -172,7 +172,7 @@ def handler(ex):
             t_error = get_sdk(self.command_loader.cli_ctx,
                               ResourceType.DATA_STORAGE,
                               'common._error#AzureHttpError')
-            if isinstance(ex, t_error) and ex.status_code == 403:
+            if isinstance(ex, t_error) and ex.status_code == 403 and ex.error_code == 'AuthorizationPermissionMismatch':
```

Comment:
Upon narrowing `ex.error_code` to `AuthorizationPermissionMismatch` and `AuthorizationFailure`, if the key or connection string is incorrect, `ex.error_code` will be `AuthenticationFailed` which is not handled.